### DISCUSSION TITLE
Check a Json object has any null values

### DISCUSSION THREAD

**Discussion by naderafshari** [nlohmann/json#3304] [Q&A] [answered]
What is the best/easiest way to check if a json object has values of null?
For example the string dump looks like this
{"eventTime":"2022-01-29T06:08:04.001709Z","id":null,"source":"svc:testmanager.hez.vpc-carson"}

Is there a way to check the whole object to see if there are any null values?
If not, what is the best way to go through the each key and check its value for null value?
I tried this
```
    Json j = Json::object();
    j = Json::parse(jsonString);
    std::cout << "id = " << j["id"] << std::endl;
    auto it = j.find("id");
    std::cout << "*it = " << *it << std::endl;
``` 
and I get this 
```
id = null
*it = null
```
but I don't know how to use it, say for example, in an if statement?
```
if (*it)
```
won't work. The error is...
```
[json.exception.type_error.302] type must be boolean, but is null
```

**Comment by nlohmann** [comment]
https://json.nlohmann.me/api/basic_json/is_null/

**Comment by naderafshari** [comment, marked as answer]
Thank you but how do I use is_null()? Seems it returns true if the entire object is null not if it has elements of null value. 
On the same object mentioned above this code returns true
```
    if (j.is_null())
    {
        std::cout << "json has null elements" << std::endl;
        return false;
    }
    std::cout << "json has no null elements" << std::endl;
    return true;
```
Is there a way to loop through elements of an object and use is_null and individual elements

**Reply by nlohmann** [reply]
You can use std::any_of.

**Reply by naderafshari** [reply]
Thank you so much it worked
```
    if (std::any_of(j.begin(), j.end(), [](Json j){return j.is_null();}))
    {
        return false;
    }
    return true;
```

**Reply by naderafshari** [reply]
It however does not pick up on nulls in nested objects. It probably needs another level of checking. Is there a way to tell if an object is nested?
For example if an object is like this...
```
{
    "id":"1-567535",
    "data":
    {
        "admin":"Locked",
        "configured":null
     }
}
```

**Reply by nlohmann** [reply]
For recursive checks, you can use code like https://stackoverflow.com/a/46630394/266378

**Reply by naderafshari** [reply]
Got it to work with that link. Thanks again
